Classes/Hooks/PageRenderer/TrackingCodeInjector.php

```diff
@@ -81,7 +81,7 @@ public function execute(?array &$params, PageRenderer $pageRenderer): void
 
     private function hasValidConfiguration(Configuration $configuration): bool
     {
-        if (! \filter_var($configuration->url, \FILTER_VALIDATE_URL)) {
+        if (empty($configuration->url) || parse_url($configuration->url) === false) {
```

Comment:
`parse_url()` won't work this way:
see https://3v4l.org/QYojn

"On seriously malformed URLs, parse_url() may return false"
https://www.php.net/manual/en/function.parse-url.php#refsect1-function.parse-url-returnvalues

Reply:
Maybe something like this beforehand:

    $urlToCheck = (str_starts_with('//', $configuration->url) ? 'https:' : '') . $configuration->url;

and then the check with `filter_var()` against `$urlToCheck`;

Reply:
That would defeat the purpose since it wouldn't allow Matomo to run on http://. This is the case e.g. in our local development environment.

Reply:
I meant only for the check, the url itself is untouched.